sources/packages/backend/apps/api/src/route-controllers/restriction/_tests_/e2e/restriction.aest.controller.addInstitutionRestriction.e2e-spec.ts

```diff
@@ -0,0 +1,434 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import {
+  E2EDataSources,
+  RestrictionCode,
+  createE2EDataSources,
+  createFakeInstitution,
+  createFakeUser,
+  createFakeEducationProgram,
+  saveFakeInstitutionRestriction,
+} from "@sims/test-utils";
+import {
+  AESTGroups,
+  BEARER_AUTH_TYPE,
+  createFakeLocation,
+  createTestingAppModule,
+  getAESTToken,
+  getAESTUser,
+} from "../../../../testHelpers";
+import * as request from "supertest";
+import {
+  EducationProgram,
+  Institution,
+  InstitutionLocation,
+  NoteType,
+  Restriction,
+  User,
+} from "@sims/sims-db";
+import { In } from "typeorm";
+import { INSTITUTION_RESTRICTION_ALREADY_ACTIVE } from "../../../../constants";
+
+describe("RestrictionAESTController(e2e)-addInstitutionRestriction.", () => {
+  let app: INestApplication;
+  let db: E2EDataSources;
+  let sharedAuditUser: User;
+  let susRestriction: Restriction;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+    sharedAuditUser = await db.user.save(createFakeUser());
+    // Find the SUS restriction to be used as an example of a valid restriction.
+    susRestriction = await db.restriction.findOne({
+      select: { id: true },
+      where: {
+        restrictionCode: RestrictionCode.SUS,
+      },
+    });
+  });
+
+  it("Should add multiple institution restrictions when a valid payload with multiple locations IDs is submitted.", async () => {
+    // Arrange
+    const [institution, program, locationIds, [location1, location2]] =
```

Comment:
Minor and not blocker. Returning locationIds seems to redundant when both locations are returned.